cylc/flow/dbstatecheck.py

```diff
@@ -67,34 +69,51 @@ def __init__(self, rund, workflow, db_path=None):
             raise OSError(errno.ENOENT, os.strerror(errno.ENOENT), db_path)
         self.conn = sqlite3.connect(db_path, timeout=10.0)
 
+        # Get workflow point format.
+        try:
+            self.point_fmt = self._get_pt_fmt()
+            self.back_compat_mode = False
+        except sqlite3.OperationalError as exc:
+            # BACK COMPAT: Cylc 7 DB (see method below).
+            try:
+                self.point_fmt = self._get_pt_fmt_compat()
+                self.back_compat_mode = True
+            except sqlite3.OperationalError:
+                raise exc  # original error
+
     @staticmethod
     def display_maps(res):
         if not res:
             sys.stderr.write("INFO: No results to display.\n")
         else:
             for row in res:
-                sys.stdout.write((", ").join(row) + "\n")
+                sys.stdout.write((", ").join([str(s) for s in row]) + "\n")
 
-    def get_remote_point_format(self):
+    def _get_pt_fmt(self):
```

Comment:
How about `_get_scheduler_cycle_point_format` - it's a mouthful, but it then exactly matches the config item, the value of which are are getting from the DB.

Reply:
As a private method now a self-explanatory name is a bit less import, but I'll consider the options...